internal/store/bigtable/reader.go

```diff
@@ -76,72 +76,62 @@ func readRowFn(
 }
 
 // Read reads BigTable rows from multiple Bigtable in parallel.
-//
-// Reading multiple rows is chunked as the size limit for RowSet is 500KB.
-//
-// Args:
-// baseBt: The bigtable that holds the base cache
-// rowSet: BigTable rowSet containing the row keys.
-// action: A callback function that converts the raw bytes into appropriate
-//		go struct based on the cache content.
-// getToken: A function to get back the indexed token (like place dcid) from
-//		bigtable row key.
-//
+// Note all Bigtable read use the same set of rowList.
 func Read(
 	ctx context.Context,
 	btGroup *Group,
-	rowSet cbt.RowSet,
+	rowSet cbt.RowList,
 	action func([]byte) (interface{}, error),
 	getToken func(string) (string, error),
 ) ([]map[string]interface{}, error) {
-	tables := btGroup.Tables()
-	if len(tables) == 0 {
-		return nil, status.Errorf(
-			codes.NotFound, "Bigtable instance is not specified")
+	rowSetMap := map[int]cbt.RowList{}
+	for i := 0; i < len(btGroup.Tables()); i++ {
+		rowSetMap[i] = rowSet
 	}
+	return ReadWithGroupRowSet(ctx, btGroup, rowSetMap, action, getToken)
+}
 
-	// Function start
-	var rowSetSize int
-	var rowList cbt.RowList
-	var rowRangeList cbt.RowRangeList
-	switch v := rowSet.(type) {
-	case cbt.RowList:
-		rowList = rowSet.(cbt.RowList)
-		rowSetSize = len(rowList)
-	case cbt.RowRangeList:
-		rowRangeList = rowSet.(cbt.RowRangeList)
-		rowSetSize = len(rowRangeList)
-	default:
-		return nil, status.Errorf(codes.Internal, "Unsupported RowSet type: %v", v)
+// ReadWithGroupRowSet reads BigTable rows from multiple Bigtable in parallel.
+// Reading is chunked as the size limit for RowSet is 500KB.
+//
+// Note the read could have different RowList for each import group Bigtable as
+// needed by the pagination APIs.
+func ReadWithGroupRowSet(
+	ctx context.Context,
+	btGroup *Group,
+	rowSetMap map[int]cbt.RowList,
```

Comment:
Is it worthwhile to change the variable name: rowSetMap->rowListMap?

Reply:
Done